build.yaml

```diff
@@ -2110,6 +2110,8 @@ libs:
   - grpc
   filegroups:
   - grpcpp_channelz_proto
+  extra_cond:
```

Comment:
This seems pretty adhoc, and also in the long run we'll be getting rid of build.yaml, so it doesn't make sense to add new properties like this.
We could just specialcase directly in CMakeLists.txt.template if needed, but I'm not sure it's worth doing it considering that gRPC_USE_PROTO_LITE is probably used only by few.